components/post_view/post_list/post_list.tsx

```diff
@@ -37,110 +38,124 @@ function markAndMeasureChannelSwitchEnd(fresh = false) {
+         * Used to loading posts since a timestamp to sync the posts
          */
-        changeUnreadChunkTimeStamp: PropTypes.func.isRequired,
+        syncPostsInChannel: (channelId: string, since: number, prefetch: boolean) => Promise<void>;
 
         /*
-         * Used for skipping the call on load
+         * Used to loading posts if it not first visit, permalink or there exists any postListIds
+         * This happens when previous channel visit has a chunk which is not the latest set of posts
          */
-        isPrefetchingInProcess: PropTypes.bool.isRequired,
-
-        isMobileView: PropTypes.bool.isRequired,
-
-        actions: PropTypes.shape({
-
-            /*
-             * Used for getting permalink view posts
-             */
-            loadPostsAround: PropTypes.func.isRequired,
-
-            /*
-             * Used for geting unreads posts
-             */
-            loadUnreads: PropTypes.func.isRequired,
+        loadLatestPosts: (channelId: string) => Promise<void>;
 
-            /*
-             * Used for getting posts using BEFORE_ID and AFTER_ID
-             */
-            loadPosts: PropTypes.func.isRequired,
+        markChannelAsViewed: (channelId: string) => void;
 
-            /*
-             * Used to set mobile view on resize
-             */
-            checkAndSetMobileView: PropTypes.func.isRequired,
-
-            /*
-             * Used to loading posts since a timestamp to sync the posts
-             */
-            syncPostsInChannel: PropTypes.func.isRequired,
-
-            /*
-             * Used to loading posts if it not first visit, permalink or there exists any postListIds
-             * This happens when previous channel visit has a chunk which is not the latest set of posts
-             */
-            loadLatestPosts: PropTypes.func.isRequired,
-
-            markChannelAsViewed: PropTypes.func.isRequired,
+        markChannelAsRead: (channelId: string) => void;
+        updateNewMessagesAtInChannel: typeof updateNewMessagesAtInChannel;
+    };
+}
 
-            markChannelAsRead: PropTypes.func.isRequired,
-            updateNewMessagesAtInChannel: PropTypes.func.isRequired,
+interface State {
+    loadingNewerPosts: boolean;
+    loadingOlderPosts: boolean;
+    autoRetryEnable: boolean;
+}
 
-        }).isRequired,
+export default class PostList extends React.PureComponent<Props, State> {
+    private autoRetriesCount: number;
+    private actionsForPostList: {
+        loadOlderPosts: () => Promise<void>;
+        loadNewerPosts: () => Promise<void>;
+        checkAndSetMobileView: () => void;
+        canLoadMorePosts: (type: CanLoadMorePosts) => Promise<void>;
+        changeUnreadChunkTimeStamp: (lastViewedAt?: string) => void;
+        updateNewMessagesAtInChannel: typeof updateNewMessagesAtInChannel;
     }
+    private mounted: boolean | undefined;
+
+    // public for testing purposes only
+    public extraPagesLoaded: number | undefined;
```

Comment:
Sounds good to me. For this change, I was able to add that repo as a remote and cherry-pick the change. I'd imagine the other will go first, but in case mine does, I'll make sure to add a co-authored by in the commit log so that the original author still gets credit. If the same is doable for the other PR, I'll do that too.